test/functional/Java20andUp/src/org/openj9/test/jep425/VirtualThreadTests.java

```diff
@@ -0,0 +1,313 @@
+ *******************************************************************************/
+package org.openj9.test.jep425;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import org.testng.AssertJUnit;
+import static org.testng.Assert.fail;
+
+import java.lang.Thread;
+import java.lang.reflect.*;
+import java.time.Duration;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import java.util.concurrent.locks.LockSupport;
+
+/**
+ * Test cases for JEP 425: Virtual Threads (Preview) Continuation execution
+ * which verifies the basic cases including Continuation enter, yield, resume.
+ */
+@Test(groups = { "level.sanity" })
+public class VirtualThreadTests {
+	static {
+		try {
+			System.loadLibrary("j9ben");
+		} catch (UnsatisfiedLinkError e) {
+			System.out.println("No natives for JNI tests");
+		}
+	}
+
+	public static native boolean lockSupportPark();
+
+	@Test
+	public void test_basicVirtualthread() {
+		var wrapper = new Object(){ boolean executed = false; };
+		try {
+			Thread t = Thread.ofVirtual().name("duke").unstarted(() -> {
+				wrapper.executed = true;
+			});
+
+			t.start();
+			t.join();
+
+			AssertJUnit.assertTrue("Virtual Thread operation not executed", wrapper.executed);
+		} catch (Exception e) {
+			Assert.fail("Unexpected exception occured : " + e.getMessage() , e);
+		}
+	}
+
+	@Test
+	public void test_VirtualthreadYieldResume() {
+		try (var executor = Executors.newVirtualThreadPerTaskExecutor()) {
+			int numThreads = 6;
+			int expectedThreadResult = 3;
+			long estimatedThreadCompletionTime = 5L; /* seconds */
+			int[] results = new int[numThreads];
+
+			IntStream.range(0, numThreads).forEach(i -> {
+					executor.submit(() -> {
+							results[i] = 1;
+							Thread.sleep(Duration.ofSeconds(1));
+							results[i] += 1;
+							Thread.sleep(Duration.ofSeconds(1));
+							results[i] += 1;
+							return i;
+					});
+			});
+
+			/* Wait incrementally for the worst-case scenario where all virtual threads are
+			 * executed sequentially. Exit the wait loop if the virtual threads finish early.
+			 */
+			for (int i = 0; i < numThreads; i++) {
+				executor.awaitTermination(estimatedThreadCompletionTime, TimeUnit.SECONDS);
+				boolean exit = true;
+				for (int j = 0; j < numThreads; j++) {
+					if (results[j] != expectedThreadResult) {
+						exit = false;
+					}
+				}
+				if (exit) {
+					break;
+				}
+			}
+
+			for (int i = 0; i < numThreads; i++) {
+				AssertJUnit.assertTrue(
+						"Virtual Thread " + i + ": incorrect result of " + results[i],
+						(results[i] == expectedThreadResult));
+			}
+		} catch (Exception e) {
+			Assert.fail("Unexpected exception occured : " + e.getMessage() , e);
+		}
+	}
+
+	private static volatile boolean testSyncThread_ready = false;
```

Comment:
Please remove the redundant initializer.

Reply:
Removed.